Make naming of 'getXxxById' consistent across the project

barroij opened the pull request:
The majority of occurrence is `getXxxByID` but there are a few `getXxxById`. This PR purpose is to always use `getXxxByID`

That's not a big deal and it's breaking backward compatibility, so it might not be worth it. But if it has to be fixed, a major version like 5.0 might a the good time to do it.

So it's up to you guys @deltakosh @RaananW & Co :)

Just close it if you don't want of it

RaananW commented:
Well, that makes total sense :-)

I am actually more for `getXXXById` and less for `byID`. especially since the browser uses `Id` and not `ID` (i.e. `getElementById`). And it is possible to have a proxy to old functions, something along the lines of:

```javascript
public get getXXXByID() {
  return this.getXXXById;
}
```

This way we keep consistency, backwards compatibility, and standards :-)

Popov72 commented:
And you can tag the old ones with something like `/** @deprecated Please use getXXXById instead */`

barroij commented:
@RaananW @Popov72 
I made the changes almost everywhere.

These ones are let unchanged : 
`Ligth.getTypeID`
`SpriteMap.getTileID` 
`InstancesBlock.instanceID`
`ICustomAnimationFrameRequester.requestID`

When the change was a public property, I changed them and added a deprecated get/set proxy (even for hidden props).

RaananW commented on gui/src/2D/xmlLoader.ts:
```diff
@@ -289,6 +289,11 @@ export class XmlLoader {
     }
 
+    /** @deprecated Please use getNodeById instead */
+     public getNodeByID(id: string): any {
```
Is that needed?

barroij replied:
you tell me :) I can remove it

RaananW replied:
you just added it, no? there is no need to add new functions, just change existing ones

barroij replied:
yes you're right, I put myself in automatic mode and add a few unwanted deprecated function 👎 .

RaananW commented on src/PostProcesses/postProcess.ts:
```diff
@@ -884,7 +884,7 @@ export class PostProcess {
         }
 
-        var camera = scene ? scene.getCameraByID(parsedPostProcess.cameraId) : null;
+        var camera = scene ? scene.getCameraByI(parsedPostProcess.cameraId) : null;
```
missing a `d`

RaananW commented on src/XR/features/WebXRImageTracking.ts:
```diff
@@ -150,6 +150,11 @@ export class WebXRImageTracking extends WebXRAbstractFeature {
     }
 
+    /** @deprecated Please use getTrackedImageById instead */
+    public getTrackedImageByID(id: number): Nullable<IWebXRTrackedImage> {
```
probably not needed?

RaananW commented on src/scene.ts:
```diff
@@ -2737,12 +2757,17 @@ export class Scene extends AbstractScene implements IAnimatable, IClipPlanesHold
      * @returns the camera or null if not found
      */
-    public getCameraByID(id: string): Nullable<Camera> {
+    public getCameraByI(id: string): Nullable<Camera> {
```
missing a `d`

RaananW commented on src/scene.ts:
```diff
@@ -2752,12 +2777,17 @@ export class Scene extends AbstractScene implements IAnimatable, IClipPlanesHold
      * @returns the camera or null if not found
      */
-    public getCameraByUniqueID(uniqueId: number): Nullable<Camera> {
+    public getCameraByUniqueI(uniqueId: number): Nullable<Camera> {
```
missing a `d`

RaananW reviewed (changes requested):
Thanks! just a few small changes

barroij commented:
@RaananW I fixed the missing  `d`s and removed the changes that were not needed  (xmlLoader.ts & ebXRImageTracking.ts )

RaananW commented:
> @RaananW I fixed the missing `d`s and removed the changes that were not needed (xmlLoader.ts & WebXRImageTracking.ts )

I might have missed one - are the ones in viewerManager required?

barroij commented:
@RaananW it should be ok now, I removed the ones in viewerManager. sorry for that 😕

RaananW commented on src/Behaviors/Meshes/pointerDragBehavior.ts:
```diff
@@ -34,10 +34,18 @@ export class PointerDragBehavior implements Behavior<AbstractMesh> {
      */
     public _useAlternatePickedPointAboveMaxDragAngle = false;
+    /** @deprecated Please use currentDraggingPointerId instead */
+    public get currentDraggingPointerID(): number {
```
Are these needed? I don't see the change of the existing function.

RaananW commented on src/Behaviors/Meshes/pointerDragBehavior.ts:
```diff
@@ -34,10 +34,18 @@ export class PointerDragBehavior implements Behavior<AbstractMesh> {
      * The id of the pointer that is currently interacting with the behavior (-1 when no pointer is active)
      */
-    public currentDraggingPointerID = -1;
+    public currentDraggingPointerId = -1;
```
This changes back-compat, requires getter.

RaananW commented on src/Behaviors/Meshes/sixDofDragBehavior.ts:
```diff
@@ -40,10 +40,18 @@ export class SixDofDragBehavior implements Behavior<Mesh> {
      */
     public dragDeltaRatio = 0.2;
+    /** @deprecated Please use currentDraggingPointerId instead */
+    public get currentDraggingPointerID(): number {
```
Same - are these needed?

RaananW commented on src/Behaviors/Meshes/sixDofDragBehavior.ts:
```diff
@@ -40,10 +40,18 @@ export class SixDofDragBehavior implements Behavior<Mesh> {
      * The id of the pointer that is currently interacting with the behavior (-1 when no pointer is active)
      */
-    public currentDraggingPointerID = -1;
+    public currentDraggingPointerId = -1;
```
Same as above - back-compat

RaananW reviewed (changes requested):
Sorry, went over it again and have a few changes.
It's a big semantic change, we want to be sure back-compat is guaranteed.

There is also a small doc issue

barroij commented:
@RaananW I'm closing this one and make another branch in which I'll check the changes one by one.
